File: JAVASCRIPT/1045.js

//https://www.urionlinejudge.com.br/judge/en/problems/view/1045
var input = require('fs').readFileSync('./input.txt', 'utf8');
var lines = input.split('\n');

const [x, y, z] = lines[0].split(" ").map(num => Number(num));

let a, b, c;
if (x > y && x > z) {
    a = x
    if (y > z) {
        b = y
        c = z
    } else {
        b = z
        c = y
    }
} else if (y > x && y > z) {
    a = y
    if (x < z) {
        b = z
        c = x
    } else {
        b = x
        c = z
    }
} else {
    a = z
    if (y > x) {
        b = y
        c = x
    } else {
        b = x
        c = y
    }
}
if (a >= b + c) {
    console.log("NAO FORMA TRIANGULO");
} else {
    if ((a * a) == (b * b) + (c * c)) {
        console.log("TRIANGULO RETANGULO");
    }
    if ((a * a) > (b * b) + (c * c)) {
        console.log("TRIANGULO OBTUSANGULO");
    }
    if ((a * a) < (b * b) + (c * c)) {
        console.log("TRIANGULO ACUTANGULO");
    }
    if (a === b && b === c) {
        console.log("TRIANGULO EQUILATERO");
    }
    if ((a == b && a !== c) || (a == c && a !== b) || (c == a && a !== b) || (c == b && c !== a)) {
        console.log("TRIANGULO ISOSCELES");
    }
}

File: C99/beginner/1045.c
#include <stdio.h>

int main() {
  double lado1, lado2, lado3;  // entradas
  double a, b, c;   // ondem decrescente
  double A, B, C;   //potencias

  scanf("%lf %lf %lf", &lado1, &lado2, &lado3);

  // arranjando os lados em ordem descrente
  if (lado1 >= lado2 && lado1 >= lado3) {
    a = lado1;
    if (lado2 >= lado3) {
      b = lado2;
      c = lado3;
    } else {
      b = lado3;
      c = lado2;
    }
  } else if (lado2 >= lado1 && lado2 >= lado3) {
    a = lado2;
    if (lado1 >= lado3) {
      b = lado1;
      c = lado3;
    } else {
      b = lado3;
      c = lado1;
    }
  } else if (lado3 >= lado1 && lado3 >= lado2) {
    a = lado3;
    if (lado1 >= lado2) {
      b = lado1;
      c = lado2;
    } else {
      b = lado2;
      c = lado1;
    }
  } else {
    a = lado1;
    b = lado2;
    c = lado3;
  }

  // potencias
  A = a * a;
  B = b * b;
  C = c * c;

  // se nao formar triangulo nao pode haver mais validações
  if (a >= b + c) {
    printf("NAO FORMA TRIANGULO\n");
  } else {
    // se formar triangulo
    if (A == B + C) {
      printf("TRIANGULO RETANGULO\n");
    }

    if (A > B + C) {
      printf("TRIANGULO OBTUSANGULO\n");
    }

    if (A < B + C) {
      printf("TRIANGULO ACUTANGULO\n");
    }

    if (a == b && b == c) {
      printf("TRIANGULO EQUILATERO\n");
    }

    if ((a == b && a != c) || (a == c && a != b) || (c == b && c != a)) {
      printf("TRIANGULO ISOSCELES\n");
    }
  }
  return 0;
}